test-utils/src/lib.rs

```diff
@@ -293,3 +297,34 @@ pub fn build_expected_hash(values: &[u64]) -> [Felt; 4] {
 
     expected
 }
+
+/// Mutates a seed and generates a word deterministically
+#[cfg(not(target_family = "wasm"))]
+pub fn seeded_word(seed: &mut [u8; 32]) -> Word {
+    [
+        seeded_element(seed),
+        seeded_element(seed),
+        seeded_element(seed),
+        seeded_element(seed),
+    ]
+}
+
+/// Mutates a seed and generates an element deterministically
+#[cfg(not(target_family = "wasm"))]
+pub fn seeded_element(seed: &mut [u8; 32]) -> Felt {
+    *seed = rand::prng_array(*seed);
+    let bytes = rand::prng_array(*seed);
+    let num = u64::from_le_bytes(bytes);
+    Felt::new(num)
+}
+
+/// Returns a remaining path key for a Sparse Merkle tree
+#[cfg(not(target_family = "wasm"))]
+pub fn get_remaining_key(mut key: Word, depth: u8) -> Word {
+    key[3] = Felt::new(match depth {
+        16 | 32 | 48 => (key[3].as_int() << depth) >> depth,
+        64 => 0,
+        _ => unreachable!(),
+    });
+    key
+}
```

Comment:
I wouldn't place this function in the root module since it is Merkle tree specific. Probably a better place for it would be in the [crypto](https://github.com/0xPolygonMiden/miden-vm/blob/next/test-utils/src/crypto.rs) module.

Also, why is this not turned on for `wasm` targets?

Reply:
Regarding the module; fixed!

Regarding the `wasm`, the reason is

https://github.com/0xPolygonMiden/miden-vm/blob/1226528a3f39a29915a32b396f3481c46e51ed35/test-utils/Cargo.toml#L27-L29

As consequence of

https://github.com/facebook/winterfell/blob/9b33ffe63ce48385349381f25341111376de19bd/utils/rand/src/lib.rs#L11-L28

I opted to not touch this WASM question and only continue the previously set pattern. However, we might want to revisit this question in the future as, I think, this problem of the past for rand with WASM is currently solved - we just need to pick the better option for winterfell

Reply:
Ah - yes! Makes sense.

Reply:
Regarding `wasm` - I don't think this function uses any random values - so, I don't think removing this flag should break anything.